src/main/python/pgshovel/replication/streams/kafka.py

```diff
@@ -40,12 +54,44 @@ def consume(self, state):
             consumer.seek(delta, 1)
             assert consumer.offsets[0] == offset
 
-        for offset, message in consumer:
-            decoded = self.codec.decode(message.value)
-            state = validate_state(state, offset, decoded)
+        for offset, message in self.__primed(consumer, state):
```

Comment:
So I actually did think of this situation but I researched and thought that since I am not overwriting the `state` here during priming, that when we skipped over the invalid batch to the next `BeginOperation`, that the validation would still fail as the sequence number did not increase by 1, but increased by the number of messages that were skipped?  Was that incorrect?

That all said, I think only priming when there is no existing stream state is a more correct and safer algorithm, as it doesn't rely on other validation or skip messages when there is existing stream state.